src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -86,31 +88,83 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_stored_table_node(
 std::shared_ptr<AbstractOperator> LQPTranslator::_translate_predicate_node(
     const std::shared_ptr<AbstractLQPNode>& node) const {
   const auto input_operator = translate_node(node->left_child());
-  auto table_scan_node = std::dynamic_pointer_cast<PredicateNode>(node);
+  auto predicate_node = std::dynamic_pointer_cast<PredicateNode>(node);
 
-  const auto column_id = table_scan_node->get_output_column_id(table_scan_node->column_reference());
+  const auto column_id = predicate_node->get_output_column_id(predicate_node->column_reference());
 
-  auto value = table_scan_node->value();
+  auto value = predicate_node->value();
   if (is_lqp_column_reference(value)) {
-    value = table_scan_node->get_output_column_id(boost::get<const LQPColumnReference>(value));
+    value = predicate_node->get_output_column_id(boost::get<const LQPColumnReference>(value));
   }
 
+  if (predicate_node->scan_type() == ScanType::IndexScan)
+    return _translate_predicate_node_to_index_scan(predicate_node, value, column_id, input_operator);
```

Comment:
Multi-line if without braces. Can we get the linter to complain about that?

Reply:
Ah, I would not introduce them, but I don't think of the formatter breaking long lines. Seems like cpplint does not support it.